lib/core/src/Cardano/Wallet.hs

```diff
@@ -671,9 +698,9 @@ listAddresses ctx wid = db & \DBLayer{..} -> do
     -- delegation addresses. So we normalize them all to be delegation addresses
     -- to make sure that we compare them correctly.
     normalize :: k 'AddressK XPub -> Address -> Maybe Address
-    normalize rewardAccount addr = do
+    normalize rewardAcc addr = do
```

Comment:
**[change required]**

Fix the top-level signature by removing `s ~ SeqState n k` and using `hasRewardAccount s` instead

**Why**

As the `FIXME` mentions, this was an ugly hack that would better be tackled by using a better abstraction.

Reply:
`rewardAccount` returns a `Maybe ChimericAccount`, and we need a `k 'AddressK XPub` here.

We could change `rewardAccount` to be `Maybe (k 'AddressK XPub)`, but we'd need to involve `k` in the `HasRewardAccount s` class… 🤔